mlflow/metrics/base.py

```diff
@@ -76,18 +77,34 @@ class EvaluationExample:
     output: str
     score: float
     justification: str
-    grading_context: Dict[str, str] = None
+    grading_context: Optional[Union[str, Dict[str, str]]] = None
+
+    @staticmethod
+    def _format_grading_context(grading_context):
+        return "\n".join([f"key: {key}\nvalue:\n{value}" for key, value in grading_context.items()])
+
+    def to_prompt_component(self, grading_context_columns: List[str]) -> str:
```

Comment:
Should we still use `__str__` for cases where the grading_context is a dict and we have all the information?
That way, users can actually read the example string inline as an output of this class?

Reply:
I think for the usecase where the its a string, we can use 

```
"prompt": evaluation_context["eval_prompt"].format(
                    input=input, output=output, grading_context_columns=arg_string
                ),
``` 
with some special key to fill in that information directly here(genai_metric).

Just adding another way to approach this problem. Thoughts?

Reply:
Great idea - addressed!